internal/executor/cache.go

```diff
@@ -350,7 +351,8 @@ func UploadCache(executor *Executor, commandName string, cacheHost string, instr
 		// check if some other task has uploaded the cache already
 		response, _ := httpClient.Head(fmt.Sprintf("http://%s/%s", cacheHost, cache.Key))
 		if response != nil && response.StatusCode == http.StatusOK {
-			logUploader.Write([]byte(fmt.Sprintf("\nSome other task has already uploaded cache entry %s! Skipping upload...", cache.Key)))
+			createdByTaskId := response.Header.Get(http_cache.CirrusHeaderCreatedBy)
```

Comment:
Wouldn't this be always empty when uploading to cache servers other than the one built into the agent?

Reply:
Unlikely but will add a check for an empty header.

Reply:
See 80422f4